packages/cli/src/util/createServer.ts

```diff
@@ -37,27 +39,22 @@ const createMultiProcessPrism: CreatePrism = async options => {
     return;
   } else {
     const logInstance = createLogger('CLI', cliSpecificLoggerOptions);
-    try {
-      return createPrismServerWithLogger(options, logInstance);
-    } catch (e) {
+
+    return createPrismServerWithLogger(options, logInstance).catch(e => {
       logInstance.fatal(e.message);
       cluster.worker.kill();
-    }
+    });
   }
 };
 
-const createSingleProcessPrism: CreatePrism = async options => {
+const createSingleProcessPrism: CreatePrism = options => {
   signale.await({ prefix: chalk.bgWhiteBright.black('[CLI]'), message: 'Starting Prism…' });
 
   const logStream = new PassThrough();
   const logInstance = createLogger('CLI', cliSpecificLoggerOptions, logStream);
   pipeOutputToSignale(logStream);
 
-  try {
-    return createPrismServerWithLogger(options, logInstance);
-  } catch (e) {
-    logInstance.fatal(e.message);
-  }
+  return createPrismServerWithLogger(options, logInstance).catch(e => logInstance.fatal(e.message));
```

Comment:
Not sure whether `logInstance.fatal` returns any value, probably not. But that: https://github.com/stoplightio/prism/blob/master/packages/cli/src/util/runner.ts#L11 relies on this function returning a server instance or undefined. So if it gets a thing it will assume it is a server instance. I would:
```suggestion
  return createPrismServerWithLogger(options, logInstance).catch(e => { logInstance.fatal(e.message); });
```

I would also investigate the option of returning rejected promise. It would remove some magic from the code.

Reply:
Good point. After the log I'll rethrow the exception.

Reply:
The same should be done here: https://github.com/stoplightio/prism/pull/1107/files#diff-986568f87b3f6dba35678ad3af18cf90R45 ?

Also, I have a bad feeling that this will break something. Somebody intentionally did not used promises :/

Reply:
@karol-maciaszek Can you point to the specific line? I can't see it

Reply:
Yes, line 45 same file. `createMultiProcessPrism()` seems to be doing exactly the same.